automation/nightly/test-nightly-build.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -ex
+
+# Get golang
+docker login --username "$(cat "${DOCKER_USER}")" --password-stdin < "${DOCKER_PASSWORD}"
+wget -q https://dl.google.com/go/go1.15.2.linux-amd64.tar.gz
+tar -C /usr/local -xf go*.tar.gz
+export PATH=/usr/local/go/bin:$PATH
+
+# get latest commits
+latest_kubevirt=$(curl -sL https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/latest)
+latest_kubevirt_image=$(curl -sL "https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${latest_kubevirt}/kubevirt-operator.yaml" | grep 'OPERATOR_IMAGE' -A1 | tail -n 1 | sed 's/.*value: //g')
+IFS=: read -r kv_image kv_tag <<< "${latest_kubevirt_image}"
+latest_kubevirt_commit=$(curl -sL "https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${latest_kubevirt}/commit")
+
+# Update HCO dependencies
+go mod edit -require "kubevirt.io/kubevirt@${latest_kubevirt_commit}"
+go mod vendor
+rm -rf kubevirt
+
+# Get latest kubevirt
+git clone https://github.com/kubevirt/kubevirt.git
+(cd kubevirt; git checkout "${latest_kubevirt_commit}")
+go mod edit -replace kubevirt.io/client-go=./kubevirt/staging/src/kubevirt.io/client-go
+go mod vendor
+go mod tidy
+
+# set envs
+build_date="$(date +%Y%m%d)"
+export IMAGE_REGISTRY=docker.io
+export IMAGE_TAG="${build_date}_$(git show -s --format=%h)"
+export DOCKER_PREFIX=kubevirtnightlybuilds
+TEMP_OPERATOR_IMAGE=${DOCKER_PREFIX}/hyperconverged-cluster-operator
+TEMP_WEBHOOK_IMAGE=${DOCKER_PREFIX}/hyperconverged-cluster-webhook
+sed -i "s#docker.io/kubevirt/virt-#${kv_image/-*/-}#" deploy/images.csv
+sed -i "s#^KUBEVIRT_VERSION=.*#KUBEVIRT_VERSION=\"${kv_tag}\"#" hack/config
+sed -i "s#quay.io/kubevirt/hyperconverged-cluster-operator#${TEMP_OPERATOR_IMAGE}#" deploy/images.csv
+sed -i "s#quay.io/kubevirt/hyperconverged-cluster-webhook#${TEMP_WEBHOOK_IMAGE}#" deploy/images.csv
+sed -i "s#^CSV_VERSION=.*#CSV_VERSION=\"${IMAGE_TAG}\"#" hack/config
+(cd ./tools/digester && go build .)
+./automation/digester/update_images.sh
+
+# Build HCO & HCO Webhook
+make container-build-operator container-push-operator container-build-webhook container-push-webhook
```

Comment:
I think we got this reversed - we need to first build build the operator and webhook, push them, and then update the images

Reply:
This is how it done in project-infra, and also, there is no real dependency here, since we're building the images witha known name, we can use this name here.

Reply:
No, we first build and push the operator, then we run the digester and build-manifests

https://github.com/kubevirt/project-infra/blob/62be3285c7f9f5700d3643b02e104c4d8ca03a82/github/ci/prow/files/jobs/kubevirt/hyperconverged-cluster-operator/hyperconverged-cluster-operator-periodics.yaml#L54-L61